### DISCUSSION TITLE
Host PocketBase on a separate server with respect to my SvelteKit backend

### DISCUSSION THREAD

**Discussion by ivan94fi** [pocketbase/pocketbase#1903] [Q&A]
Hi,

I am wandering if it is possible to host the backend (in my case a SvelteKit app) in a separate server wrt the pocketbase executable.

From the docs, and all the available examples I found, I can see that the client is always created with a local url, for example:
```javascript
import PocketBase from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

...
```

It is not clear to me if the backend code using the javascript SDK can be separated from the PocketBase installation, for exemple by hosting the two services in different machines on the Internet, then connecting the machines.

Is this use case viable? My biggest concern is how I could make the backend code and PocketBase communicate securely.

Maybe a related discussion is #930.

Thank you.

**Comment by ganigeorgiev** [comment]
I'm not entirely sure that I understand what you are asking.

The url in the SDK client needs to point to your PocketBase application (it can be deployed separately from your SvelteKit app, even on a different host).

The exposed PocketBase APIs are secured by the collections API rules. If you haven't already, you can find more info in https://pocketbase.io/docs/api-rules-and-filters/.

For a SvelteKit SSR example and general JS SDK integration info, you can check https://github.com/pocketbase/js-sdk#ssr-integration.

**Comment by ivan94fi** [comment]
Hi, thank you for the quick reply.

After further research I found exactly what I ultimately intended to do, which is what is specified in [this feature request](https://github.com/pocketbase/pocketbase/issues/718), that is, being able to authenticate with just a user-specific token (a permanent token) generated from the admin. This would enable our application to make authenticated requests programmatically, as our app is actually an automatic service, not a user-facing app.

As an example, see [this link](https://firebase.google.com/docs/admin/setup#initialize_the_sdk_in_non-google_environments) in Firebase documentation: a "service account key file" is created via the Firebase admin console, then the generated token can be used to authenticate services which cannot access user credentials.

At this point my new question is: do you plan to support this feature any soon?

Thank you for your work!

**Reply by ganigeorgiev** [reply]
You can always just authenticate as admin instead of using a fixed token:

```js
await pb.admins.authWithPassword("test@example.com", "123456")
...
```

For the currently planned tasks and their priority you can refer to https://github.com/orgs/pocketbase/projects/2.

**Comment by ivan94fi** [comment]
> You can always just authenticate as admin instead of using a fixed token:

That is true, but I would loose the authorization this way: that is, any action could be performed on the database, there would be no protection. Instead, if using an api token specific for each user the authorizations for that user would be kept.

Thank you for your answers, I will certainly start to use PocketBase once #718 is implemented!
